### PR TITLE
rename mysql err to common format

### PR DESCRIPTION
**Pull request by siddontang**

### CODE REVIEW

**Comment by qiuyesuifeng** [comment]
LGTM

File: mysqldef/error_test.go [hunk 1]
```diff
@@ -23,13 +23,13 @@ type testSQLErrorSuite struct {
...
 func (s *testSQLErrorSuite) TestSQLError(c *C) {
-	e := NewError(ErNoDbError, "no db error")
+	e := NewError(ErrNoDbError, "no db error")
 	c.Assert(len(e.Error()), Greater, 0)
 
 	e = NewError(0, "customized error")
 	c.Assert(len(e.Error()), Greater, 0)
 
-	e = NewDefaultError(ErNoDbError)
+	e = NewDefaultError(ErrNoDbError)
```
Comment (ngaut) [inline mysqldef/error_test.go]: s/ErrNoDbError/ErrNoDb

File: mysqldef/state.go [hunk 2]
```diff
@@ -21,229 +21,229 @@ const (
...
-	ErAlterOperationNotSupported:          "0A000",
-	ErAlterOperationNotSupportedReason:    "0A000",
-	ErDupUnknownInIndex:                   "23000",
+	ErrDupKey:                              "23000",
+	ErrOutofmemory:                         "HY001",
+	ErrOutOfSortmemory:                     "HY001",
+	ErrConCountError:                       "08004",
+	ErrBadHostError:                        "08S01",
+	ErrHandshakeError:                      "08S01",
+	ErrDbaccessDeniedError:                 "42000",
```
Comment (ngaut) [inline mysqldef/state.go]: s/ErrDbaccessDeniedError/ErrDbaccessDenied

Comment (ngaut) [inline mysqldef/state.go]: s/ErrUnknownComError/ErrUnknownCom

File: mysqldef/state.go [hunk 3]
```diff
@@ -21,229 +21,229 @@ const (
...
+	ErrAccessDeniedError:                   "28000",
+	ErrNoDbError:                           "3D000",
+	ErrUnknownComError:                     "08S01",
+	ErrBadNullError:                        "23000",
+	ErrBadDbError:                          "42000",
+	ErrTableExistsError:                    "42S01",
```
Comment (ngaut) [inline mysqldef/state.go]: s/ErrTableExistsError/ErrTableExists

**Comment by siddontang** [comment]
PTAL

**Pull request merged**